### ISSUE TITLE
Promise support

### ISSUE THREAD

**Issue by Richienb** [electerious/nice-try#25]
It would be useful if nice-try supported promises:

```js
const niceTry = require('nice-try')

const result = await niceTry.promise(async () => {...})
```

Implementation:

```js
module.exports.promise = async function(fn) {

	try { return await fn() } catch (e) {}

}
```

// @electerious

**Comment by Ongshu777** [comment]
Maybe this should be PR instead of an issue; you already have the implementation.

**Comment by electerious** [comment]
Just published v2.1 🚀
